app/src/routes/editor/WebviewArea/BrowserControl.tsx

```diff
@@ -32,6 +34,14 @@ function BrowserControls({
     setHovered,
     onlookEnabled,
 }: BrowserControlsProps) {
+    const editorEngine = useEditorEngine();
+    const [scale, setScale] = useState(1);
+
+    useEffect(() => {
+        setScale(editorEngine.canvas.scale);
+        console.log('print', editorEngine.canvas.scale);
+        console.log(scale);
+    }, [editorEngine.canvas]);
```

Comment:
`editorEngine.canvas.scale` should be the value you're listening to.

Reply:
This will detect when canvas is reassigned which should never happen. It prints if you listen to the scale value here instead

```
    useEffect(() => {
        setScale(editorEngine.canvas.scale);
        console.log('print', editorEngine.canvas.scale);
        console.log(scale);
    }, [editorEngine.canvas.scale]);
```